GameServer.cpp

```diff
@@ -308,19 +308,18 @@ void GameServer::ProcessPoke(Packet& packet, sockaddr_in& from)
 			counters.numFailedGameInfoAllocs++;
 			return;			// Abort  (failed to allocate new record)
 		}
-		GameInfo* currentGameInfo = &gameInfos[gameInfoIndex];
 
 		LogEndpoint("Game Hosted from: ", from.sin_addr.s_addr, from.sin_port);
 
-		// Initialize the new record
-		currentGameInfo->addr = from;
-		currentGameInfo->clientRandValue = packet.tlMessage.gameServerPoke.randValue;
-		currentGameInfo->serverRandValue = GetNewRandValue();
-		currentGameInfo->flags |= GameInfoExpected;
-		currentGameInfo->time = time(0);
+		GameInfo& newGameInfo = gameInfos[gameInfoIndex];
+		newGameInfo.addr = from;
+		newGameInfo.clientRandValue = packet.tlMessage.gameServerPoke.randValue;
+		newGameInfo.serverRandValue = GetNewRandValue();
+		newGameInfo.flags |= GameInfoExpected;
+		newGameInfo.time = time(0);

```

Comment:
This feels strangely like an allocation without a `push_back`. Granted, there is no counter adjustment code being removed, so this is probably fine. It just seems a bit non-intuitive. My guess is, it's a result of poor structure in the old code, which separated allocation from initialization. We may want to examine this part in a future PR.

Reply:
I'm not fully sure what is happening. It would be helpful to me if we documented the `PokeStatusCode` enum members.